platforms/jvm/testing-jvm/src/main/java/org/gradle/api/tasks/testing/TestReport.java

```diff
@@ -95,10 +102,26 @@ public abstract class TestReport extends DefaultTask {
     @TaskAction
     void generateReport() {
         try {
-            TestReportGenerator impl = detectAndCreateImplementation(getTestResults());
-            if (impl.hasResults()) {
-                impl.generateReport(getBuildOperationRunner(), getBuildOperationExecutor(), getDestinationDirectory().get().getAsFile().toPath());
-            } else {
+            List<Path> resultDirsAsPaths = new ArrayList<>(getTestResults().getFiles().size());
+            Boolean isGenericImplementation = null;
+            for (File resultDir : getTestResults().getFiles()) {
+                boolean resultDirIsGenericImplementation = SerializableTestResultStore.isGenericTestResults(resultDir);
+                if (isGenericImplementation == null) {
+                    isGenericImplementation = resultDirIsGenericImplementation;
+                } else if (isGenericImplementation != resultDirIsGenericImplementation) {
+                    throw new IllegalStateException("Cannot mix generic and non-generic test results in the same report.");
+                }
+                resultDirsAsPaths.add(resultDir.toPath());
+            }
```

Comment:
🤔 This is probably worth re-extracting to a `private boolean isGenericResultsDir(FileCollection testResults)` method.  Why merge this into a the single `generateReport` method?